src/legacy/file-save.mjs

```diff
@@ -43,4 +43,5 @@ export default async (blob, options = {}) => {
     setTimeout(() => URL.revokeObjectURL(a.href), 30 * 1000);
   });
   a.click();
+  return null;
```

Comment:
```suggestion
  return Promise.resolve(null);
```

Not a TypeScript expert, but I think it should be the above, according to a quick [test](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwBQEoBciAKAnOAWxgGcBTAHijxDMQB9EwQAbFgPkQG8AoRfxHjJQQeJPiKkyAOiEk4LAG5lUzNugDcPAL48eoSLASIARhmwTi5KjTqM1HbnwFCRYpqxZbdQA).

Reply:
Hey there. Thank you for the review!

Your test accidentally omitted the `async` keyword. This function is asynchronous so [any result is promisified automatically](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#description) so I think this is ok as-is. For example:

```js
(async function() { return 1; })() instanceof Promise // true
(async function() { })() instanceof Promise // true
(async function() { return null; })() instanceof Promise // true
await (async function() { return 1; })() // 1
await (async function() { })() // undefined
await (async function() { return null; })() // null
```

Compare to the same synchronous functions:

```js
(function() { return 1; })() instanceof Promise // false
(function() { })() instanceof Promise // false
(function() { return null; })() instanceof Promise // false
await (function() { return 1; })() // 1
await (function() { })() // undefined
await (function() { return null; })() // null
```

Reply:
Perfect then, thanks for catching this!